source-stripe-native/source_stripe_native/api.py

```diff
@@ -0,0 +1,495 @@
+    http: HTTPSession,
+    log: Logger,
+    page: str | None,
+    cutoff: datetime,
+)-> AsyncGenerator:
+    """Note: Stripe's data is always served in reverse-chronological order.
+    fetch_backfill_substreams works similar to fetch_backfill. The only variation is that 
+    the resulting data serves has the search object for the next
+    stream (child stream). With that, a new iteration happens inside
+    this method exclusively for the child stream.
+    """
+
+    _cls: Any = cls  # Silence mypy false-positive
+
+    url = f"{API}/{_cls.SEARCH_NAME}"
+    parameters = {"limit": 100}
+
+    search_name = _cls.SEARCH_NAME
+
+
+    if page:
+        parameters["starting_after"] = page
+    
+    if _cls.NAME == "Subscriptions" or cls.NAME == "SubscriptionItems":
+        parameters["status"] = "all"
+
+
+    
+    result = BackfillResult[_cls].model_validate_json(
+        await http.request(log, url, method="GET", params=parameters)
+    )
+
+    for doc in result.data:
+        log.debug(f"{_s_to_dt(doc.created)}")
+        if _s_to_dt(doc.created) == stop_date:
+            parent_data = doc
+            id = parent_data.id
+            child_data = _capture_substreams(
+                            cls_child,
+                            search_name,
+                            id,
+                            parent_data,
+                            http,
+                            log
+                        )
+
+            if child_data is None:
+                return
+            async for doc in child_data:
+                doc.parent_id = parent_data.id
+                doc.meta_ = cls_child.Meta(op="u")
+                yield doc 
+            return
+
+        elif _s_to_dt(doc.created) < stop_date:
+            return
+
+        elif _s_to_dt(doc.created) < cutoff:
+            parent_data = doc
+            id = parent_data.id
+            child_data = _capture_substreams(
+                            cls_child,
+                            search_name,
+                            id,
+                            parent_data,
+                            http,
+                            log
+                        )
+
+            if child_data is None:
+                pass # move to next customer
+            async for doc in child_data:
+                doc.parent_id = parent_data.id
+                doc.meta_ = cls_child.Meta(op="u")
+                yield doc 
+
+    if result.has_more:
+        yield result.data[-1].id
+    else:
+        return
+
+async def _capture_substreams(
+    cls_child,
+    search_name,
+    id,
+    parent_data,
+    http,
+    log
+):
+    """_capture_substreams works by handling the child_stream query and pagination.
+    it requires the parent stream data, along with the item id and the parent search name.
+    """
+
+    child_url = f"{API}/{search_name}/{id}/{cls_child.SEARCH_NAME}"
+    parameters = cls_child.PARAMETERS
+    if "subscription" in parameters.keys():
```

Comment:
The body of this function looks pretty ugly. Lots of magic strings and branching. I don't understand the Stripe data model well enough with child streams etc. or whatever is going on here to offer any suggestion other than that. I'd suggest trying to clean this up.

Reply:
This was to handle some special cases and dont make the models be that much different from one another. The `subscription`  one was actually removed now, and the check was changed to compare only the stream name. 
Mod was made here 00467212d8135d9c433b5806262295c55c368f4c